streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/AbstractSinkTask.java

```diff
@@ -84,28 +85,32 @@ public void put(Collection<SinkRecord> records) {
       LOG.debug("ConnectorSinkTask: {} sink records received", records.size());
     }
 
-    List<T> eventsToPublish = new ArrayList<>();
-    for (SinkRecord record : records) {
-      T dataToPublish = getDataToPublish(SinkUtils.getStringPayload(GSON, record));
-      eventsToPublish.add(dataToPublish);
-      bytesRead += getPayloadLength(dataToPublish);
+    try {
+      List<T> eventsToPublish = new ArrayList<>();
+      for (SinkRecord record : records) {
+        T dataToPublish = getDataToPublish(SinkUtils.getStringPayload(GSON, record));
+        eventsToPublish.add(dataToPublish);
+        bytesRead += getPayloadLength(dataToPublish);
+
+        long tempCurrTime = System.currentTimeMillis();
+        if (flushNow(tempCurrTime)) {
+          publishAndLogIfRequired(eventsToPublish);
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("ConnectorSinkTask: {} events flushed partially", eventsToPublish.size());
+          }
+          reset(eventsToPublish, tempCurrTime);
+        }
+      }
 
-      long tempCurrTime = System.currentTimeMillis();
-      if (flushNow(tempCurrTime)) {
+      if (!eventsToPublish.isEmpty()) {
         publishAndLogIfRequired(eventsToPublish);
```

Comment:
try catch could be added on publishAndLogIfRequired method since this method is sending message to AEP and exception could be catch there and ConnectException is thrown for 500 and unauthorised errors.

Reply:
Ok done